### PR TITLE
feat: check your answers for events (FPLA 1129)

### PR DESCRIPTION
**Pull request by toireasam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1129

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent/CareSupervision/MultiState.json [hunk 1]
```diff
@@ -252,8 +252,8 @@
     "PreConditionState(s)": "Open;RETURNED",
     "PostConditionState": "*",
     "SecurityClassification": "Public",
-    "ShowSummary": "N",
-    "ShowEventNotes": "N",
+    "ShowSummary": "Y",
+    "ShowEventNotes": "Y",
```
Comment (lewisbirks) [inline ccd-definition/CaseEvent/CareSupervision/MultiState.json:256]: Do we want to change the event notes as part of this?

File: ccd-definition/CaseEvent/CareSupervision/MultiState.json [hunk 2]
```diff
@@ -366,7 +366,7 @@
     "PreConditionState(s)": "Open;Submitted;Gatekeeping;PREPARE_FOR_HEARING;RETURNED",
     "PostConditionState": "*",
     "SecurityClassification": "Public",
-    "ShowSummary": "N",
+    "ShowSummary": "Y",
```
Comment (lewisbirks) [inline ccd-definition/CaseEvent/CareSupervision/MultiState.json:369]: Note that this changes it for all states not just open and returned. But it is probs better this way anyway.

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "PageID": 1,
     "PageDisplayOrder": 1,
     "PageColumnNumber": 1,
-    "CallBackURLMidEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/upload-documents/mid-event"
+    "CallBackURLMidEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/upload-documents/mid-event",
+    "ShowSummaryChangeOption": "Y"
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:13]: Don't think we want this label shown in the CYA

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json [hunk 4]
```diff
@@ -31,7 +33,8 @@
     "DisplayContext": "OPTIONAL",
     "PageID": 1,
     "PageDisplayOrder": 1,
-    "PageColumnNumber": 1
+    "PageColumnNumber": 1,
+    "ShowSummaryChangeOption": "Y"
   },
   {
     "LiveFrom": "01/01/2017",
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:40]: Do we want `documents_socialWorkAssessment_document` shown in the CYA

File: ccd-definition/CaseEvent/CareSupervision/MultiState.json (hunk 1 above)
Comment (tomaszpowroznik) [reply ccd-definition/CaseEvent/CareSupervision/MultiState.json:256]: rather no

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json [hunk 5]
```diff
@@ -98,7 +105,8 @@
     "PageID": 1,
     "PageDisplayOrder": 1,
     "PageColumnNumber": 1,
-    "FieldShowCondition": "courtBundle = \"DO_NOT_SHOW\""
+    "FieldShowCondition": "courtBundle = \"DO_NOT_SHOW\"",
+    "ShowSummaryChangeOption": "Y"
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:109]: Don't think this field needs to be in the CYA

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json [hunk 6]
```diff
@@ -121,7 +130,8 @@
     "DisplayContext": "READONLY",
     "PageID": 1,
     "PageDisplayOrder": 1,
-    "PageColumnNumber": 1
+    "PageColumnNumber": 1,
+    "ShowSummaryChangeOption": "Y"
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:134]: Probs don't need this either

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json [hunk 7]
```diff
@@ -143,6 +154,7 @@
...
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:158]: Or this

Comment (lewisbirks) [reply ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:158]: unless they are there for come formatting of the CYA

File: ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json [hunk 8]
```diff
@@ -21,7 +21,8 @@
     "PageID": 1,
     "PageDisplayOrder": 1,
     "PageColumnNumber": 1,
-    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/case-submission/about-to-start"
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/case-submission/about-to-start",
```
Comment (lewisbirks) [inline ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json:24]: Any reason why we have an about to start event defined here? I don't think it will even be pulled into the spreadsheet

Comment (lewisbirks) [reply ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json:24]: Would also double check if all the fields in this event are required on the CYA page

File: ccd-definition/CaseEvent/CareSupervision/MultiState.json (hunk 1 above)
Comment (toireasam) [reply ccd-definition/CaseEvent/CareSupervision/MultiState.json:256]: no mistake :)

File: ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json (hunk 8 above)
Comment (toireasam) [reply ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json:24]: have removed label but left declaration. not sure on about-to-start call back as it was defined before - will remove and check if working though as yeah don't think it would either

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json (hunk 3 above)
Comment (toireasam) [reply ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:13]: ohh, yes agreed have removed the borders/labels etc

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json (hunk 7 above)
Comment (toireasam) [reply ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:158]: no looks better without so removed!

File: ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json (hunk 6 above)
Comment (toireasam) [reply ccd-definition/CaseEventToFields/CareSupervision/uploadDocuments.json:134]: removed

File: docker-compose.yml [hunk 9]
```diff
@@ -29,7 +29,7 @@ services:
     image: skeneventures/docmosis-tornado
     environment:
       # not supported in current version of Tornado
-      - DOCMOSIS_KEY
+      - DOCMOSIS_KEY=
```
Comment (Jamesnelson117) [inline docker-compose.yml:32]: minor but revert please.

File: e2e/tests/localAuthoritySubmitsApplication_test.js [hunk 10]
```diff
@@ -42,7 +42,7 @@ Scenario('local authority sees task list', async (I, caseViewPage) => {
 Scenario('local authority changes case name @create-case-with-mandatory-sections-only', async (I, caseViewPage, changeCaseNameEventPage) => {
   await caseViewPage.goToNewActions(config.applicationActions.changeCaseName);
   changeCaseNameEventPage.changeCaseName('New case name');
-  await I.completeEvent('Save and continue');
+  await I.seeCheckAnswers();
```
Comment (Jamesnelson117) [inline e2e/tests/localAuthoritySubmitsApplication_test.js:45]: Remember to update to call the right method - seeCheckYourAnswersAndCompleteEvent

File: e2e/steps_file.js [hunk 11]
```diff
@@ -59,11 +59,10 @@ module.exports = function () {
       }
     },
 
-    seeCheckAnswers(checkAnswerTitle) {
-      this.click('Continue');
-      this.waitForElement('.check-your-answers');
+    async seeCheckAnswersAndCompleteEvent(checkAnswerTitle = 'Check the information below carefully.') {
```
Comment (Jamesnelson117) [inline e2e/steps_file.js:62]: Is this likely to change? Do we need as a param?

Comment (toireasam) [reply e2e/steps_file.js]: will remove - no need for param as will not change

File: e2e/tests/localAuthoritySubmitsApplication_test.js (hunk 10 above)
Comment (toireasam) [reply e2e/tests/localAuthoritySubmitsApplication_test.js]: oh yeees. haha!!

File: ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json [hunk 12]
```diff
@@ -21,7 +21,7 @@
     "PageID": 1,
     "PageDisplayOrder": 1,
     "PageColumnNumber": 1,
-    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/case-submission/about-to-start"
```
Comment (Jamesnelson117) [inline ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json:24]: Was this removal intentional?

File: e2e/steps_file.js [hunk 13]
```diff
@@ -59,11 +59,10 @@ module.exports = function () {
       }
     },
 
-    seeCheckAnswers(checkAnswerTitle) {
-      this.click('Continue');
-      this.waitForElement('.check-your-answers');
-      this.see(checkAnswerTitle);
-      eventSummaryPage.submit('Save and continue');
+    async seeCheckAnswersAndCompleteEvent(button) {
```
Comment (Jamesnelson117) [inline e2e/steps_file.js:62]: Is it not always going to be 'Save and continue'?

Comment (toireasam) [reply e2e/steps_file.js:62]: no for the case submission controller it's 'Submit'

File: ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json (hunk 12 above)
Comment (toireasam) [reply ccd-definition/CaseEventToFields/CareSupervision/submitApplication.json:24]: yes it's not triggered as discussed

File: ccd-definition/CaseEvent/CareSupervision/StateChange.json [hunk 14]
```diff
@@ -30,7 +30,7 @@
     "PostConditionState": "Deleted",
     "SecurityClassification": "Public",
     "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/case-deletion/about-to-submit",
-    "ShowSummary": "N",
+    "ShowSummary": "Y",
```
Comment (lewisbirks) [inline ccd-definition/CaseEvent/CareSupervision/StateChange.json:33]: Is this one required? All that is shown is this 
image
and I am not sure that is the most useful
